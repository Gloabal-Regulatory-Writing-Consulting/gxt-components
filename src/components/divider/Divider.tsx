import { HTMLAttributes } from "react";
import {
  HorizontalDividerWrapper,
  VerticalDividerWrapper,
} from "./Divider.styles";

export type DividerLayout = "horizontal" | "vertical";
export type DividerColor = "light" | "medium";

export type DividerProps = {
  layout?: DividerLayout;
  color?: DividerColor;
  dataTestId?: string;
} & HTMLAttributes<HTMLDivElement>;

function Divider({
  layout = "horizontal",
  color = "light",
  dataTestId = "divider",
  ...rest
}: DividerProps) {
  const DividerWrapper =
    layout == "horizontal" ? HorizontalDividerWrapper : VerticalDividerWrapper;
  return <DividerWrapper $color={color} {...rest} data-testid={dataTestId} />;
}

export default Divider;
