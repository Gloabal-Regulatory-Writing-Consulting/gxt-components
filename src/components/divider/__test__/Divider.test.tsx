import React from "react";
import { render, screen } from "@testing-library/react";
import { describe, expect, test } from "vitest";
import "jest-styled-components";

import Divider from "../Divider";

describe("Divider", () => {
  test("renders horizontal divider with light color", () => {
    render(<Divider layout="horizontal" color="light" />);
    expect(screen.getByTestId("divider")).toMatchSnapshot();
  });

  test("renders vertical divider with medium color", () => {
    render(<Divider layout="vertical" color="medium" />);
    expect(screen.getByTestId("divider")).toMatchSnapshot();
  });
});
