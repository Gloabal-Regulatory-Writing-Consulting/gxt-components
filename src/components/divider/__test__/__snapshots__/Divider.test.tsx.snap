// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Divider > renders horizontal divider with light color 1`] = `
.c0 {
  height: 0.0625rem;
  flex-shrink: 0;
  background-color: var(--neutral-200, #9CA3AF);
}

<div
  class="c0"
  data-testid="divider"
/>
`;

exports[`Divider > renders vertical divider with medium color 1`] = `
.c0 {
  height: 0.0625rem;
  flex-shrink: 0;
  background-color: var(--neutral-400, #414141);
}

.c1 {
  transform: rotate(90deg);
}

<div
  class="c0 c1"
  data-testid="divider"
/>
`;
